src/main/java/org/springframework/data/keyvalue/repository/config/KeyValueRepositoryConfigurationExtension.java

```diff
@@ -145,7 +145,7 @@ public void registerBeansForRoot(BeanDefinitionRegistry registry, RepositoryConf
 
 			AbstractBeanDefinition beanDefinition = getDefaultKeyValueTemplateBeanDefinition(configurationSource);
 
-			if (beanDefinition != null) {
+			if (beanDefinition != null && configurationSource.getSource() != null) {
 				registerIfNotAlreadyRegistered(() -> beanDefinition, registry, keyValueTemplateName.get(),
```

Comment:
Looks like a missing `@Nullable` annotation for `source` in Commons.

Reply:
maybe... care to check in commons.

Reply:
Fixed via https://github.com/spring-projects/spring-data-commons/commit/6e97f3bec87330a1944a9f52b15731175ab1f84e